src/Fetcher/ConfTechFetcher.php

```diff
@@ -32,136 +34,206 @@ public function __construct(RegistryInterface $doctrine, LoggerInterface $logger
-                            $tag = $this->em->getRepository(Tag::class)->findOneBy([
-                                'name' => $fetchedTag,
-                            ]);
+        return $location;
+    }
 
-                            if ($tag instanceof Tag) {
-                                $conference->addTag($tag);
-                            }
-                        }
-                    }
+    private function pushConf(array $fetchedConferences, $newConferencesCount, $source, $technologie, $conferenceTotal = [])
+    {
+        $tag = $this->getTagByName($technologie[1]);
 
-                    if (isset($fetchedConference->cfpUrl, $fetchedConference->cfpEndDate)) {
-                        $cfpEndAt = \DateTime::createFromFormat('Y-m-d', $fetchedConference->cfpEndDate);
+        foreach ($fetchedConferences as $fC) {
+            $fC = $this->hash($fC);
+            $fC->tag = $tag;
+            $fC->source = $source;
 
-                        $conference->setCfpUrl($fetchedConference->cfpUrl);
-                        $conference->setCfpEndAt($cfpEndAt);
-                        $conference1['cfp_url'] = $fetchedConference->cfpUrl;
-                        $conference1['cfp_end_at'] = $cfpEndAt;
-                    } else {
-                        $conference1['cfp_url'] = null;
-                        $conference1['cfp_end_at'] = null;
-                    }
+            $conference = $this->repository->findOneBy([
+            'hash' => $fC->hash,
+            ]);
 
-                    array_push($conferenceTotal, $conference1);
+            if (!$conference) {
+                $conference = $this->repository->findOneBy([
+                    'slug' => $fC->slug, ]);
 
-                    ++$newConferencesCount;
-                    $conference1 = [];
+                // Do not override a conference created by another source
+                if ($conference && $conference->getSource() !== $source) {
+                    continue;
                 }
+            }
+
+            if (!$conference) {
+                $conference = $this->setParams($fC);
+
+                $this->em->persist($conference);
 
-                array_push($conferencesFinal, $conferenceTotal);
+                array_push($conferenceTotal, $conference);
+                ++$newConferencesCount;
             }
         }
 
-        return $conferencesFinal;
+        return [
+            'conferenceTotal' => $conferenceTotal,
+            'newConferenceCount' => $newConferencesCount,
+        ];
     }
 
-    public function getLocation($conference)
+    private function matchTags()
     {
-        $location = $conference->city;
+        $dateCurrentYear = date('Y');
+        $dateNextYear = date('Y', strtotime('+1 year'));
+
+        $tagsSelected = $this->em->getRepository(Tag::class)->findBy(['selected' => true]);
+
+        $tagMatch = [
+            'Android' => 'android',
```

Comment:
For keys you should use Enum instead of hard rewrite tags.

Reply:
How do I know which key is it ?
Do I need to keep the same order than Tag enum ? Or in alphabetical order for both ?

Reply:
You can do `array_combine(TagEnum::toArray(), $confTechFetcherSynonyms)` where `confTechFetcherSynonyms` contains all values of `$tagMatch` array. In this case, yes the order matters.